### PR TITLE
Remove Roles Classes

### PR DESCRIPTION
**Pull request by ryanrath**
## Description
**NOTE: this PR is based on #772 and builds on it by removing all of the Role classes and their associated tests / test artifacts**

This is it, the PR we've all been waiting for ( or at least I have ). After this will just be cleanup of dead code, tables etc. that are no longer used. 

## Motivation and Context
Because these files / tests are no longer used. 

## Tests performed
All tests for docker: fresh_install && upgrade
- unit
- component
- integration
- REG_TEST_ALL=1 regression
- UI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Removing Dead code

## Checklist:

- [X] My code follows the code style of this project as found in the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: classes/DataWarehouse/Access/MetricExplorer.php [hunk 1]
```diff
@@ -617,11 +618,8 @@ public static function convertActiveRoleToGlobalFilters(XDUser $user, $activeRol
         } else {
             $activeRoleComponents = explode(':', $activeRoleId);
             $activeRoleComponents = array_pad($activeRoleComponents, 2, null);
```
Comment (plessbd) [inline classes/DataWarehouse/Access/MetricExplorer.php]: array_pad doesnt seem to be needed here

Comment (ryanrath) [reply classes/DataWarehouse/Access/MetricExplorer.php]: Yup, see: https://github.com/ubccr/xdmod/pull/772#pullrequestreview-191770960

**Comment by ryanrath** [comment]
`Roles` / `aRole` need to stay in until we finish the Configuration Code refactor as that's what provides the `extends` functionality. `iRole` may be able to be removed though. Lemme just double check real quick.

**Comment by smgallo** [comment]
`aRole` implements `iRole` so it'll need to stay for now.

**Comment by ryanrath** [comment]
It does, but the one function that `iRole` specifies is never used so it should be safe to remove `iRole` and `aRole`'s implementation of `iRole`.

**Comment by smgallo** [comment]
OK, lets leave `iRole` for now and clean it up when `aRole` goes away.

**Comment by ryanrath** [comment]
*nod* sounds good

**Pull request merged**